data_steward/tools/create_update_drc_id_match_table.py

```diff
@@ -0,0 +1,229 @@
+USING `{{project_id}}.{{drc_dataset_id}}.{{ps_values_table_id}}` a
+ON a.person_id = b.person_id
+WHEN NOT MATCHED THEN 
+INSERT (_PARTITIONTIME, {{fields}}) 
+VALUES (TIMESTAMP_TRUNC(CURRENT_TIMESTAMP, HOUR), {{fields}})
+""")
+
+MISSING_EHR_QUERY = JINJA_ENV.from_string("""
+UPDATE `{{project_id}}.{{drc_dataset_id}}.{{id_match_table_id}}`
+SET {{col}} = 'missing_ehr'
+WHERE {{col}} IS NULL
+""")
+
+MISSING_RDR_QUERY = JINJA_ENV.from_string("""
+UPDATE `{{project_id}}.{{drc_dataset_id}}.{{id_match_table_id}}`
+SET {{set_expression}}
+WHERE COALESCE({{fields}}) IS NULL
+""")
+
+SET_EXPRESSION = JINJA_ENV.from_string("""
+{{col}} = 'missing_rdr'""")
+
+
+def create_drc_validation_table(client, project_id, table_id):
+    """
+    Creates the partitioned DRC validation table. Table will be partitioned by HOUR.
+
+    :param client: bq client
+    :param project_id: the project containing the dataset
+    :param table_id: ID of the table
+    """
+
+    fields = resources.fields_for(IDENTITY_MATCH_TABLE)
+
+    create_table = CREATE_TABLE.render(project_id=project_id,
+                                       drc_dataset_id=DRC_OPS,
+                                       id_match_table_id=table_id,
+                                       fields=bq.get_bq_fields_sql(fields))
+    client.query(create_table)
+
+    LOGGER.info(f'Created {table_id}.')
+
+    return table_id
+
+
+def copy_ps_values_data_to_id_match_table(client, project_id, table_id, hpo_id):
+    """
+    Copies data from the ps_values table to the DRC identity match table.
+
+    :param client: bq client
+    :param project_id: the project containing the dataset
+    :param table_id: ID for the table
+    :param hpo_id: ID for the HPO site
+    """
+
+    schema_list = bq.get_table_schema(IDENTITY_MATCH_TABLE)
+    id_match_table_id = table_id
+    ps_values_table_id = f'{PS_API_VALUES}_{hpo_id}'
+
+    fields_name_str = ',\n'.join([item.name for item in schema_list])
+
+    insert_query = INSERT_QUERY.render(project_id=project_id,
+                                       drc_dataset_id=DRC_OPS,
+                                       id_match_table_id=id_match_table_id,
+                                       fields=fields_name_str,
+                                       ps_values_table_id=ps_values_table_id)
+
+    client.query(insert_query)
+
+    LOGGER.info(
+        f'Inserted data from `{ps_values_table_id}` to `{id_match_table_id}`')
+
+
+def get_set_expression():
+    """
+    This method generates the SET_EXPRESSION query
+    """
+    set_expression = []
+    schema_list = bq.get_table_schema(IDENTITY_MATCH_TABLE)
+    ', '.join([item.name for item in schema_list])
+
+    for item in schema_list:
+        set_expression.append(SET_EXPRESSION.render(col=item.name))
+
+    # Work around to remove person_id from the fields list since person_id is required
+    # and will never need to be updated to missing_rdr
+    set_expression.remove("\nperson_id = 'missing_rdr'")
```

Comment:
Is it possible to do this removal beforehand, maybe in the for loop? Matching this string may get finicky, especially with the newline character.